### ISSUE TITLE
Installing NexoPOS-4x

### ISSUE THREAD

**Issue by dchiputura** [Blair2004/NexoPOS-4x#255]
I have followed all the steps of how i can install NexoPos-4x on Windows from the document provided. When i run php artisan serve, it starts very well as shown below
Starting Laravel development server: http://127.0.0.1:8000
Going to the browser and type the url   http://127.0.0.1:8000 am getting a HUGE logo Image for NexoPos-4x covering the whole screen and the option buttons to Dashboard, Sign In and Sign Out appearing at the bottom left corner like shown below
If i Click on Dashboard the follow page will load with only a BIG image for NexoPOs-4x and nothing is displayed and just end there. I cannot proceed anymore with the database setup and i am stuck. What could be the problem and how can i resolve it

**If you see this page, this means NexoPOS 4.x is correctly installed on your system. As this page is meant to be the frontend, NexoPOS 4.x doesn't have a frontend for the meantime. This page shows useful links that will take you to the important resources.

Dashboard
Sign In
Sign Up** 

![image](https://user-images.githubusercontent.com/47517932/128520035-055f8bfe-6e72-424d-8db9-d678ef08a701.png)

**Comment by Blair2004** [comment]
You need to build the asset : 
```
npm run prod
```

**Comment by dchiputura** [comment]
From which directory do i run the command  npm run prod

**Comment by Blair2004** [comment]
From the root where there is the .env file.

**Comment by dchiputura** [comment]
Could you please provide me with a Step by step process of Installing NexoPos 4x on Windows. I think i have missed out on something. I would appreciate if you can share with the me the Procedure. Am still facing challenges on Installing NexoPos 4x

**Comment by Blair2004** [comment]
https://www.youtube.com/watch?v=-eXapKZrcBc

**Comment by dchiputura** [comment]
Thank you very much for the Help that you have provided so far. Actually i followed through that YouTube video in the Installation that i did. In Open Powershell window am opening the folder where I have downloaded NexoPos 4.x which is this one
C:\laragon\www\NexoPOS-4x-master
From there i am issuing these commands : composer install; php artisan key:generate; php artisan storage:link; php artisan serve
All these are executing successfully. Thats when i will go to the browser with the url http://localhost:8000 and nothing is happening as illustrated above.
I have installed node.js and by default its has installed NODE and NPM.  My question is can you explain to me further how i can built the asset because if i run  NPM RUN PROD from the command line am getting this error message:

C:\laragon\www\NexoPOS-4x-master> npm run prod

> prod
> npm run production


> production
> cross-env NODE_ENV=production node_modules/webpack/bin/webpack.js --no-progress --hide-modules --config=node_modules/laravel-mix/setup/webpack.config.js

[webpack-cli] Error: Unknown option '--no-progress'
[webpack-cli] Run 'webpack --help' to see available commands and options

**Comment by Blair2004** [comment]
Please provide a screenshot or a full output of what you get from the
terminal.

**Comment by dchiputura** [comment]
![image](https://user-images.githubusercontent.com/47517932/128554058-835f9b38-1f1d-433e-a16c-66f87ae4c78e.png)
![image](https://user-images.githubusercontent.com/47517932/128554165-7009eb0a-f4fc-42e8-ba73-035d0bdb8112.png)
![image](https://user-images.githubusercontent.com/47517932/128554253-d8f8bff7-8ad1-41ae-b341-801d81b4b206.png)
![image](https://user-images.githubusercontent.com/47517932/128554346-8343194a-0ad9-4d50-b11a-b9f5da1cfa3f.png)

**Comment by Blair2004** [comment]
Consider using the branch v4.4.x which doesn't need asset building
https://github.com/Blair2004/NexoPOS-4x/archive/refs/heads/v4.4.x.zip

**Comment by dchiputura** [comment]
Thank you very much will try that and get back to you

**Comment by dchiputura** [comment]
I have tried this new one using the same procedure from the video but still getting the same error
*Consider using the branch v4.4.x which doesn't need asset building
*https://github.com/Blair2004/NexoPOS-4x/archive/refs/heads/v4.4.x.zip

What Installation Procedure should i follow for this one which does not require asset building?

**Comment by Blair2004** [comment]
You only need to use "composer i" to install the dependencies that's all, and serve the project.

**Comment by dchiputura** [comment]
So does that mean that i need to only laragon and then install Composer and
i don't need to install node.js. Then from the on the  Powershell window
insider the folder with NaxoPos 4.x 4.4.0 i then run the following commands
compose install      and
serve

**Comment by dchiputura** [comment]
Sorry for bothering you again. Am still not getting any joy. I have now put the NaxoPos 4.x 4.4.0 folder in www folder of Laragon directory and in the Command Line have executed composer install. But still am getting these errors when i try to launch NexoPos on the browser

![image](https://user-images.githubusercontent.com/47517932/128596451-ca07ff53-de76-47a1-a5e6-079a094a24cc.png)
![image](https://user-images.githubusercontent.com/47517932/128596520-36d12109-41f5-4a79-80c6-e50bedc21d95.png)
![image](https://user-images.githubusercontent.com/47517932/128596539-82022927-aca6-4387-9a9d-df4b06801332.png)

**Comment by Blair2004** [comment]
Check on the "public" directory and if there is no storage" folder run on
the root the following command "php artisan storage:link"

**Comment by dchiputura** [comment]
The folder storage is there and inside the link folder i have the file index.html
![image](https://user-images.githubusercontent.com/47517932/128598037-c8b668a3-4293-4e3b-bbd5-2848849f1a15.png)
![image](https://user-images.githubusercontent.com/47517932/128598062-426dc547-afc2-41b3-97b8-776ab9abf16e.png)

**Comment by Blair2004** [comment]
if you haven't defined, please add a new key-pair on the ".env" file 
```
NS_ENV=production
```

**Comment by dchiputura** [comment]
I have defined it but still am getting the same errors

![image](https://user-images.githubusercontent.com/47517932/128598898-900b43fd-2a08-4512-82fb-ffcd1af6295a.png)

**Comment by Blair2004** [comment]
We'll try that from our end. We would have checked your system, but that is only reserved to premium users.
Thank you for your observations.

**Comment by dchiputura** [comment]
Am in the process of making a payment for the premium user so that you can
assist me with the installation

**Comment by dchiputura** [comment]
I have paid for the premium user please could you please check my system am really struck now

[65300201-nexopos-4x-pos-crm-inventory-manager-license.pdf](https://github.com/Blair2004/NexoPOS-4x/files/6949255/65300201-nexopos-4x-pos-crm-inventory-manager-license.pdf)

**Comment by Blair2004** [comment]
Hi, 
consider contacting us via contact@nexopos.com

**Comment by Jerad30** [comment]
I installed the Nexopos 5.3.3 local it's working fine I have the same copy of the project I uploaded into the server I configured the database and when I try to access the site the sign-in page loads but does not appear. can please advise me on what this issue is and how to solve this issue?

![Image](https://github.com/user-attachments/assets/7f28ef82-f765-4632-a08e-5705eb8b5986)

**Comment by nzian** [comment]
Please confirm that you are check all the previous comment and follow or try above comment action to solve your problem. mainly this problem found if a .env file config value is not set properly SANCTUM_STATEFUL_DOMAINS also sometime SESSION_DOMAIN value. Please check those are set properly. and make sure you have check all comment above your comment.
THank you @Jerad30

**Comment by Jerad30** [comment]
Yes, I followed the way only I installed and It's working on the Local.
after uploading to the server in the .env file I changed the domains.

this is my URL - https://textustech.dev/demopos/public/sign-in

APP_URL=textustech.dev/demopos/public
SESSION_DOMAIN=textustech.dev/demopos/public
SANCTUM_STATEFUL_DOMAINS=textustech.dev/demopos/public
REVERB_SERVER_HOST="textustech.dev/demopos/public"

**Comment by Blair2004** [comment]
> Yes, I followed the way only I installed and It's working on the Local.
> after uploading to the server in the .env file I changed the domains.
> 
> this is my URL - https://textustech.dev/demopos/public/sign-in
> 
> APP_URL=textustech.dev/demopos/public
> SESSION_DOMAIN=textustech.dev/demopos/public
> SANCTUM_STATEFUL_DOMAINS=textustech.dev/demopos/public
> REVERB_SERVER_HOST="textustech.dev/demopos/public"

This is not correct. "public" shouldn't appear on the URL. You're not doing it the right way.

[Read More](https://my.nexopos.com/en/documentation/troubleshooting/environment-misconfiguration)

**Comment by Blair2004** [comment]
"public" SHOULD not appear on the URL, it that appears on the link, you have *NOT* correctly installed NexoPOS. 

You MUST point your "[Document Root](https://www.hostgator.com/help/article/what-is-a-document-root-folder)" to the "public" folder located on NexoPOS.

[Hosting on Hostinger](https://my.nexopos.com/en/documentation/how-tos/installing-nexopos-4-x-on-hostinger)

**Comment by Jerad30** [comment]
![Image](https://github.com/user-attachments/assets/195f24ea-1af9-464e-b776-d71aded2b7c3)

I installed the systems in to the local and Uploaded in to the server now pages are loading when login the form it's showing the 
"CSRF token mismatch." Why is that?
